### ISSUE TITLE
Importing function from other file can change return type of function

### ISSUE THREAD

**Issue by Electroid** [oven-sh/bun#2027]
The following [test](https://github.com/lodash/lodash/blob/2da024c3b4f9947a48517639de7560457cd4ec6c/test/isBuffer.test.js#L4-L11) from `lodash` is failing:

```ts
// test/isBuffer.test.js
import isBuffer from '../isBuffer.js';

describe('isBuffer', function() {
  it('should return `true` for buffers', function() {
    if (Buffer) {
      assert.strictEqual(isBuffer(new Buffer(2)), true);
    }
  });
```

Output:
```ts
AssertionError: Expected values to be strictly equal:
+ actual - expected

+ { default: [Getter/Setter] }
- true
 code: "ERR_ASSERTION"

✗ isBuffer > should return `true` for buffers
```

I added some debug code to figure out what might be going wrong.

```ts
// test/isBuffer.test.js
console.log(isBuffer("test"));
```

Output:
```ts
{
  [Symbol(CommonJSTransformed)]: true
}
```

When I add the same code to the file where the function is exported, it shows a different result.

```ts
// isBuffer.js
console.log(isBuffer("test"));
```

Output:
```ts
false
```

**Comment by Jarred-Sumner** [comment]
this is an ESM <> CJS issue

**Comment by Jarred-Sumner** [comment]
fixed at runtime in v0.6
